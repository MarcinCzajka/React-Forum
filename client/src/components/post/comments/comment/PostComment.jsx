import React from 'react';
import moment from 'moment';
import { Comment } from "semantic-ui-react";
import CommentGroup from '../CommentGroup';
import CommentToolkit from '../replyForm/CommentToolkit';
import PostPlaceholder from '../../../placeholders/PostPlaceholder';
import AvatarPlaceholder from '../../../placeholders/AvatarPlaceholder';
import './PostComment.css';

class PostComment extends React.Component {
	constructor(props) {
		super(props);
		
		this.state = {
			avatar: this.props.avatar,
			avatarReady: false
		}

		this.handleReplyRef = React.createRef();
	}

	handleReplyToPost = () => {
		const { roomId, replyContent, postId } = this.props;
		
		this.handleReplyRef(roomId, replyContent, postId);
	}

	onAvatarLoad = () => {
		this.setState({avatarReady: true});
	}

	formattedDate = () => {
		return moment(this.props.date)
			.format("MMMM Do YYYY, h:mm:ss")
			.toString();	
	}
	
	render() {
		return (
			<div className="ui large comments">

				{this.props.showPlaceholder ? (
					<PostPlaceholder />
				) : (
					<Comment className="comment">

						{!this.state.avatarReady ? (
							<AvatarPlaceholder size='sizePostComment' /> 
						) : ''}

						<Comment.Avatar 
							style={{display:(this.state.avatarReady ? 'block' : 'none')}}
							className="avatar" 
							src={this.props.avatar}
							onLoad={this.onAvatarLoad}>
						</Comment.Avatar>
						
						<Comment.Content>
							<Comment.Author className="author" as='a'>{this.props.authorNick}</Comment.Author>
							<Comment.Metadata className="metadata">
								<span className="date">{this.formattedDate()}</span>
							</Comment.Metadata>
							<Comment.Text as='p' className="text">{this.props.content}</Comment.Text>

						<CommentToolkit 
							postId={this.props.postId}
							roomId={this.props.roomId}
							authorId={this.props.authorId}
							userId={this.props.userId}
							handleReply={this.handleReplyToPost}
							removeComment={this.props.removeComment}
						/>
							
						</Comment.Content>
					</Comment>
				)}

				<CommentGroup 
					parentId={this.props.postId} 
					ref={this.handleReplyRef} >
				</CommentGroup>

			</div>
		);
	}

};

export default PostComment;