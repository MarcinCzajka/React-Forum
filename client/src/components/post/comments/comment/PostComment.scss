.author {
    cursor: text !important;
    color: rgba(0,0,0,.87)  !important;
}