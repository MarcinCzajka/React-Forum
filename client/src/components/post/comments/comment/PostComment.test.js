import React from 'react';
import { shallow } from 'enzyme';
import { create } from "react-test-renderer";
import PostComment from './PostComment';
import { LocaleProvider } from '../../../../contexts/LocaleContext';
import locale from '../../../../locale/en/dictionary.json';

console.log(locale)

const roomId = "12345";
const postId = "12345";
const userId = "12345";
const authorId = "12345";
const handleReply = () => {};
const removeComment = () => {};
const avatar = "www";
let date = '2019-12-17T15:50:53.238Z';
let authorNick = 'Testotron';
let content = 'Super comment 123';

describe('PostComment component', () => {
    test('renders without crashing', () => {
        shallow(<LocaleProvider value={locale}><PostComment
            date={date}
            showPlaceholder={false}
            roomId={roomId}
            postId={postId}
            userId={userId}
            authorNick={authorNick}
            authorId={authorId}
            handleReply={handleReply}
            removeComment={removeComment}
            content={content}
        /></LocaleProvider>);
    });

    test('match snapshot while showing placeholder 1/1', () => {
        const component = create(<LocaleProvider value={locale}><PostComment 
            date={date}
            roomId={roomId}
            showPlaceholder={true}
            postId={postId}
            userId={userId}
            authorNick={authorNick}
            authorId={authorId}
            handleReply={handleReply}
            removeComment={removeComment}
            content={content}
        /></LocaleProvider>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w changing date 1/3', () => {
        const component = create(<LocaleProvider value={locale}><PostComment 
            date={date}
            roomId={roomId}
            showPlaceholder={false}
            postId={postId}
            userId={userId}
            authorNick={authorNick}
            authorId={authorId}
            handleReply={handleReply}
            removeComment={removeComment}
            content={content}
        /></LocaleProvider>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w changing date 2/3', () => {
        date = '2019-12-18T22:36:40.643Z'
        const component = create(<LocaleProvider value={locale}><PostComment 
            date={date}
            roomId={roomId}
            showPlaceholder={false}
            postId={postId}
            userId={userId}
            authorNick={authorNick}
            authorId={authorId}
            handleReply={handleReply}
            removeComment={removeComment}
            content={content}
        /></LocaleProvider>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w changing date 3/3', () => {
        date = '2019-12-20T20:18:54.783Z'
        const component = create(<LocaleProvider value={locale}><PostComment 
            date={date}
            roomId={roomId}
            showPlaceholder={false}
            postId={postId}
            userId={userId}
            authorNick={authorNick}
            authorId={authorId}
            handleReply={handleReply}
            removeComment={removeComment}
            content={content}
        /></LocaleProvider>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w changing Author Name 1/2', () => {
        authorNick = 'Deleted User.';
        const component = create(<LocaleProvider value={locale}><PostComment 
            date={date}
            roomId={roomId}
            showPlaceholder={false}
            postId={postId}
            userId={userId}
            authorNick={authorNick}
            authorId={authorId}
            handleReply={handleReply}
            removeComment={removeComment}
            content={content}
        /></LocaleProvider>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w changing Author Name 2/2', () => {
        authorNick = 'LiterallyNobody66';
        const component = create(<LocaleProvider value={locale}><PostComment 
            date={date}
            roomId={roomId}
            showPlaceholder={false}
            postId={postId}
            userId={userId}
            authorNick={authorNick}
            authorId={authorId}
            handleReply={handleReply}
            removeComment={removeComment}
            content={content}
        /></LocaleProvider>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w Avatar 1/1', () => {
        const component = create(<LocaleProvider value={locale}><PostComment 
            date={date}
            roomId={roomId}
            showPlaceholder={false}
            postId={postId}
            userId={userId}
            authorNick={authorNick}
            authorId={authorId}
            handleReply={handleReply}
            removeComment={removeComment}
            avatar={avatar}
            content={content}
        /></LocaleProvider>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w Changing Content 1/2', () => {
        content = 'very different stuff';
        const component = create(<LocaleProvider value={locale}><PostComment 
            date={date}
            roomId={roomId}
            showPlaceholder={false}
            postId={postId}
            userId={userId}
            authorNick={authorNick}
            authorId={authorId}
            handleReply={handleReply}
            removeComment={removeComment}
            avatar={avatar}
            content={content}
        /></LocaleProvider>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w Changing Content 2/2', () => {
        content = '123234tdgdsss43';
        const component = create(<LocaleProvider value={locale}><PostComment 
            date={date}
            roomId={roomId}
            showPlaceholder={false}
            postId={postId}
            userId={userId}
            authorNick={authorNick}
            authorId={authorId}
            handleReply={handleReply}
            removeComment={removeComment}
            avatar={avatar}
            content={content}
        /></LocaleProvider>);
        expect(component).toMatchSnapshot();
    });

});