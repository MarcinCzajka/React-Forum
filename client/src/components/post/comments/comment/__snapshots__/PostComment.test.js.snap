// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostComment component match snapshot /w Avatar 1/1 1`] = `
<div
  className="ui large comments"
>
  <div
    className="comment comment"
  >
    <div
      className="avatarPlaceholderContainer sizePostComment"
    >
      <i
        aria-hidden="true"
        className="user icon avatarPlaceholder"
        onClick={[Function]}
      />
    </div>
    <div
      className="avatar avatar"
      onLoad={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    >
      <img
        src="www"
      />
    </div>
    <div
      className="content"
    >
      <a
        className="author author"
      >
        LiterallyNobody66
      </a>
      <div
        className="metadata metadata"
      >
        <span
          className="date"
        >
          February 25th 2020, 9:41:55
        </span>
      </div>
      <p
        className="text text"
      >
        Super comment 123
      </p>
      <div
        className="actions"
      >
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Reply
        </button>
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
      
    </div>
  </div>
  <div
    className="ui comments recursiveComment"
  />
</div>
`;

exports[`PostComment component match snapshot /w Changing Content 1/2 1`] = `
<div
  className="ui large comments"
>
  <div
    className="comment comment"
  >
    <div
      className="avatarPlaceholderContainer sizePostComment"
    >
      <i
        aria-hidden="true"
        className="user icon avatarPlaceholder"
        onClick={[Function]}
      />
    </div>
    <div
      className="avatar avatar"
      onLoad={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    >
      <img
        src="www"
      />
    </div>
    <div
      className="content"
    >
      <a
        className="author author"
      >
        LiterallyNobody66
      </a>
      <div
        className="metadata metadata"
      >
        <span
          className="date"
        >
          February 25th 2020, 9:41:55
        </span>
      </div>
      <p
        className="text text"
      >
        very different stuff
      </p>
      <div
        className="actions"
      >
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Reply
        </button>
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
      
    </div>
  </div>
  <div
    className="ui comments recursiveComment"
  />
</div>
`;

exports[`PostComment component match snapshot /w Changing Content 2/2 1`] = `
<div
  className="ui large comments"
>
  <div
    className="comment comment"
  >
    <div
      className="avatarPlaceholderContainer sizePostComment"
    >
      <i
        aria-hidden="true"
        className="user icon avatarPlaceholder"
        onClick={[Function]}
      />
    </div>
    <div
      className="avatar avatar"
      onLoad={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    >
      <img
        src="www"
      />
    </div>
    <div
      className="content"
    >
      <a
        className="author author"
      >
        LiterallyNobody66
      </a>
      <div
        className="metadata metadata"
      >
        <span
          className="date"
        >
          February 25th 2020, 9:41:55
        </span>
      </div>
      <p
        className="text text"
      >
        123234tdgdsss43
      </p>
      <div
        className="actions"
      >
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Reply
        </button>
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
      
    </div>
  </div>
  <div
    className="ui comments recursiveComment"
  />
</div>
`;

exports[`PostComment component match snapshot /w changing Author Name 1/2 1`] = `
<div
  className="ui large comments"
>
  <div
    className="comment comment"
  >
    <div
      className="avatarPlaceholderContainer sizePostComment"
    >
      <i
        aria-hidden="true"
        className="user icon avatarPlaceholder"
        onClick={[Function]}
      />
    </div>
    <div
      className="avatar avatar"
      onLoad={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    />
    <div
      className="content"
    >
      <a
        className="author author"
      >
        Deleted User.
      </a>
      <div
        className="metadata metadata"
      >
        <span
          className="date"
        >
          February 25th 2020, 9:41:55
        </span>
      </div>
      <p
        className="text text"
      >
        Super comment 123
      </p>
      <div
        className="actions"
      >
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Reply
        </button>
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
      
    </div>
  </div>
  <div
    className="ui comments recursiveComment"
  />
</div>
`;

exports[`PostComment component match snapshot /w changing Author Name 2/2 1`] = `
<div
  className="ui large comments"
>
  <div
    className="comment comment"
  >
    <div
      className="avatarPlaceholderContainer sizePostComment"
    >
      <i
        aria-hidden="true"
        className="user icon avatarPlaceholder"
        onClick={[Function]}
      />
    </div>
    <div
      className="avatar avatar"
      onLoad={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    />
    <div
      className="content"
    >
      <a
        className="author author"
      >
        LiterallyNobody66
      </a>
      <div
        className="metadata metadata"
      >
        <span
          className="date"
        >
          February 25th 2020, 9:41:55
        </span>
      </div>
      <p
        className="text text"
      >
        Super comment 123
      </p>
      <div
        className="actions"
      >
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Reply
        </button>
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
      
    </div>
  </div>
  <div
    className="ui comments recursiveComment"
  />
</div>
`;

exports[`PostComment component match snapshot /w changing date 1/3 1`] = `
<div
  className="ui large comments"
>
  <div
    className="comment comment"
  >
    <div
      className="avatarPlaceholderContainer sizePostComment"
    >
      <i
        aria-hidden="true"
        className="user icon avatarPlaceholder"
        onClick={[Function]}
      />
    </div>
    <div
      className="avatar avatar"
      onLoad={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    />
    <div
      className="content"
    >
      <a
        className="author author"
      >
        Testotron
      </a>
      <div
        className="metadata metadata"
      >
        <span
          className="date"
        >
          February 25th 2020, 9:41:55
        </span>
      </div>
      <p
        className="text text"
      >
        Super comment 123
      </p>
      <div
        className="actions"
      >
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Reply
        </button>
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
      
    </div>
  </div>
  <div
    className="ui comments recursiveComment"
  />
</div>
`;

exports[`PostComment component match snapshot /w changing date 2/3 1`] = `
<div
  className="ui large comments"
>
  <div
    className="comment comment"
  >
    <div
      className="avatarPlaceholderContainer sizePostComment"
    >
      <i
        aria-hidden="true"
        className="user icon avatarPlaceholder"
        onClick={[Function]}
      />
    </div>
    <div
      className="avatar avatar"
      onLoad={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    />
    <div
      className="content"
    >
      <a
        className="author author"
      >
        Testotron
      </a>
      <div
        className="metadata metadata"
      >
        <span
          className="date"
        >
          February 25th 2020, 9:41:55
        </span>
      </div>
      <p
        className="text text"
      >
        Super comment 123
      </p>
      <div
        className="actions"
      >
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Reply
        </button>
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
      
    </div>
  </div>
  <div
    className="ui comments recursiveComment"
  />
</div>
`;

exports[`PostComment component match snapshot /w changing date 3/3 1`] = `
<div
  className="ui large comments"
>
  <div
    className="comment comment"
  >
    <div
      className="avatarPlaceholderContainer sizePostComment"
    >
      <i
        aria-hidden="true"
        className="user icon avatarPlaceholder"
        onClick={[Function]}
      />
    </div>
    <div
      className="avatar avatar"
      onLoad={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    />
    <div
      className="content"
    >
      <a
        className="author author"
      >
        Testotron
      </a>
      <div
        className="metadata metadata"
      >
        <span
          className="date"
        >
          February 25th 2020, 9:41:55
        </span>
      </div>
      <p
        className="text text"
      >
        Super comment 123
      </p>
      <div
        className="actions"
      >
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Reply
        </button>
        <button
          className="ui mini button"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
      
    </div>
  </div>
  <div
    className="ui comments recursiveComment"
  />
</div>
`;

exports[`PostComment component match snapshot while showing placeholder 1/1 1`] = `
<div
  className="ui large comments"
>
  <div
    className="ui placeholder"
  >
    <div
      className="image header"
    >
      <div
        className="line"
      />
    </div>
    <div
      className="paragraph"
    >
      <div
        className="line"
      />
      <div
        className="line"
      />
    </div>
  </div>
  <div
    className="ui comments recursiveComment"
  />
</div>
`;
